Patch https://github.com/calebccff/lv_drivers/pull/5, to be integrated
properly into lv_drivers.git / unl0kr upstream. Add it as separate patch
for now as this is a workaround for the very annoying "can't type fast"
bug and it's v23.12 release day.

From 36d981a7d4dff6d288c4294cf0763d7c642a2bff Mon Sep 17 00:00:00 2001
From: jane400 <pmos@j4ne.de>
Date: Sat, 16 Dec 2023 20:38:33 +0100
Subject: [PATCH] libinput: release keyboard keys in lvgl directly after
 pressing

This enables fast typing on hardware keyboards as keys are no longer forgotten.
---
 indev/libinput.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/indev/libinput.c b/indev/libinput.c
index a78419f..0f63a58 100644
--- a/indev/libinput.c
+++ b/indev/libinput.c
@@ -677,6 +677,19 @@ static void read_keypad(libinput_drv_state_t *state, struct libinput_event *even
       if (evt->key_val != 0) {
         /* Only record button state when actual output is produced to prevent widgets from refreshing */
         evt->pressed = (key_state == LIBINPUT_KEY_STATE_RELEASED) ? LV_INDEV_STATE_REL : LV_INDEV_STATE_PR;
+
+        // just release the key immediatly after it got pressed.
+        // but don't handle special keys where holding a key makes sense
+        if (evt->key_val != LV_KEY_BACKSPACE &&
+            evt->key_val != LV_KEY_UP &&
+            evt->key_val != LV_KEY_LEFT &&
+            evt->key_val != LV_KEY_RIGHT &&
+            evt->key_val != LV_KEY_DOWN &&
+            key_state == LIBINPUT_KEY_STATE_PRESSED) {
+          libinput_lv_event_t *release_evt = new_event(state);
+          release_evt->pressed = LV_INDEV_STATE_REL;
+          release_evt->key_val = evt->key_val;
+        }
       }
       break;
     default:
-- 
2.43.0

